sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/NonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Token credential class for accepting non-Azure OpenAI credential string.
+ */
+public class NonAzureOpenAIKeyCredential implements TokenCredential {
```

Comment:
This class should be moved to `models` package.

Reply:
I am not sure what the purpose of the pack `models` is, but generally I would think we should only have classes that are either used for requests or responses. Is that not necessarily the case?

Reply:
Talked offline. Decided to move to /models package as aligned to other existing SDKs.